src/main/python/hydra/zmqtest/runtest.py

```diff
@@ -26,42 +28,111 @@ def __init__(self, argv):
             config_fn = argv[1]
             del argv[1]
         RunTestBase.__init__(self, 'zmqScale', config, config_fn)
-
-        zstpub = '/zst-pub'
-        zstsub = '/zst-sub'
-        self.add_appid(zstpub)
-        self.add_appid(zstsub)
+        self.zstpub = '/zst-pub'
+        self.zstsub = '/zst-sub'
+        self.add_appid(self.zstpub)
+        self.add_appid(self.zstsub)
+        #self.total_sub_apps = 200
+        self.total_sub_apps = zmq_config.total_apps
         self.start_init()
+        self.run_test()
+
+    def run_test(self):
+        # Launch zmq pub
+        self.launch_zmq_pub()
+
+        # Launch zmq sub up to self.total_sub_apps
+        self.launch_zmq_sub()
+
+        # Signal PUB to start sending all messages, blocks until PUB notifies
+        l.info("Waiting for 10 secs before signalling PUB to send messages ..")
+        time.sleep(10)
+        self.signal_pub_send_msgs()
+
+        #  Fetch all client SUB data
+        self.fetch_all_sub_clients_data()
+
+        #  Delete all launched apps
+        self.delete_all_launched_apps()
+        l.info("Successfully finished gathering all data")
+
+        l.info("================================================")
+        for client, info in self.all_sub_clients_info.items():
+            l.info("Client: %s", client)
+            l.info("Info: %s", info)
+        l.info("================================================")
+        l.info("Please press ctrl+z  and then \"kill %1\"  twice.. sorry, will figure this out soon")
+        return
+
+    def launch_zmq_pub(self):
         l.info("Launching the pub app")
-        self.create_hydra_app(name=zstpub, app_path='hydra.zmqtest.zmqtests.zmq_pub', app_args='1555',
+        self.create_hydra_app(name=self.zstpub, app_path='hydra.zmqtest.zmq_pub.run',
+                              app_args='0.0001', # msg delay, msg count
                               cpus=0.01, mem=32,
+                              ports=[0],
                               constraints=[self.app_constraints(field='hostname', operator='UNIQUE')])
+        self.wait_app_ready(self.zstpub, 1)
+        # Find launched pub server's ip, rep PORT
+        self.pub_ip = self.find_ip_uniqueapp(self.zstpub)
+        tasks = self.get_app_tasks(self.zstpub)
+        assert(len(tasks) == 1)
+        assert(len(tasks[0].ports) == 1)
+        self.pub_rep_taskport = str(tasks[0].ports[0])
+        l.info("[zmq_pub] ZMQ pub server running at [%s]", self.pub_ip)
+        l.info("[zmq_pub] ZMQ REP server running at [%s:%s]", self.pub_ip, self.pub_rep_taskport)
+        time.sleep(2)
```

Comment:
not a particular reason, wanted some room before launching all subs. will remove